tests/common/dualtor/dual_tor_mock.py

```diff
@@ -362,12 +363,43 @@ def apply_mux_cable_table_to_dut(rand_selected_dut, mock_server_base_ip_addr, to
     dut.shell_cmds(cmds=cmds)
 
 
+def is_t0_mocked_dualtor(tbinfo):
+    return tbinfo["topo"]["type"] == "t0" and 'dualtor' not in tbinfo["topo"]["name"]
+
+
+@pytest.fixture(scope='module')
+def apply_dualtor_subtype_to_th2(duthosts):
```

Comment:
This field already gets added in `apply_peer_switch_table_to_dut`, is it possible to just add the SWSS restart functionality to that fixture in case of TH2 ASIC?

Reply:
Thanks. I didn't realise the ```subtype``` has been added in ```apply_peer_switch_table_to_dut```. 
@gechiang Could you help to confirm if it's no hurt to add ```subtype: DualToR``` for all platform, and does ```dualToR``` equal to ```DualToR```? Thanks

Reply:
@bingwang-ms the entry in `apply_peer_switch_table_to_dut` is supposed to be `DualToR`, looks like I made a typo when I first wrote it 😁 would appreciate if you could fix that for me lol

I believe the `subtype` field is needed by orchagent for all platforms, @prsunny can you confirm?

Reply:
@bingwang-ms 
The subtype field if you are matching with case sensitive then it needs to match "DualToR".  
I did a quick check on what may be impacted if subtype: "DualToR" is added blindly:

The following check for DUALTOR and perform Stopping MUX contaniner:
./src/sonic-utilities/scripts/reboot:SUBTYPE=$(sonic-cfggen -d -v DEVICE_METADATA.localhost.subtype)  

The following during config Reload ARP/FDB entries are cached if DualToR:
./src/sonic-utilities/config/main.py:    cache_arp_table = not disable_arp_cache and 'subtype' in localhost_metadata and localhost_metadata['subtype'].lower() == 'dualtor'

The following check for DualToR and Mux container is optionally enabled:
../files/build_templates/init_cfg.json.j2:{%- if include_mux == "y" %}{% do features.append(("mux", "{% if 'subtype' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['subtype'] == 'DualToR' %}enabled{% else %}always_disabled{% endif %}", false, "enabled")) %}{% endif %}

When DHCP docker started, if DualToR is configured, it enable some special options:
./dockers/docker-dhcp-relay/docker-dhcp-relay.supervisord.conf.j2:{% if 'subtype' in DEVICE_METADATA['localhost'] and DEVICE_METADATA['localhost']['subtype'] == 'DualToR' %} -U Loopback0 -dt{% endif -%}

So if your testcase is not involve in any of the above and you always restore it back (remove the subtype setting if it was set by your testcase only) then it is ok to set it for all platforms.

Reply:
> @bingwang-ms the entry in `apply_peer_switch_table_to_dut` is supposed to be `DualToR`, looks like I made a typo when I first wrote it 😁 would appreciate if you could fix that for me lol
> 
> I believe the `subtype` field is needed by orchagent for all platforms, @prsunny can you confirm?

Sure. I will corretc it.

Reply:
Updated. Thanks